### PR TITLE
[TF FE] Refactor LinSpace translator and add layer test

### PR DESCRIPTION
**Pull request by rkazants**
**Details:** Refactor LinSpace translator and add layer test

**Ticket:** 92790

Signed-off-by: Kazantsev, Roman <roman.kazantsev@intel.com>

### CODE REVIEW

File: tests/layer_tests/tensorflow_tests/test_tf_LinSpace.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            stop = tf.compat.v1.placeholder(tf.float32, start_shape, 'stop')
+            tf.raw_ops.LinSpace(start=start, stop=stop, num=num_value)
+            tf.compat.v1.global_variables_initializer()
+
+            tf_net = sess.graph_def
+
+        return tf_net, None
+
+    test_data_basic = [
+        dict(start_shape=[], num_value=2),
```
Comment (popovaan) [inline tests/layer_tests/tensorflow_tests/test_tf_LinSpace.py:24]: Probably we can add some more complex shapes?

File: tests/layer_tests/tensorflow_tests/test_tf_LinSpace.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import tensorflow as tf
+from common.tf_layer_test_class import CommonTFLayerTest
+
+
+class TestLinSpace(CommonTFLayerTest):
+    def create_lin_space_net(self, start_shape, num_value):
+        tf.compat.v1.reset_default_graph()
+        # Create the graph and model
+        with tf.compat.v1.Session() as sess:
+            start = tf.compat.v1.placeholder(tf.float32, start_shape, 'start')
```
[the remark this answers is not in the document]
Comment (rkazants) [reply tests/layer_tests/tensorflow_tests/test_tf_LinSpace.py:14]: No, only floating-point is supported: https://www.tensorflow.org/api_docs/python/tf/raw_ops/LinSpace

File: tests/layer_tests/tensorflow_tests/test_tf_LinSpace.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            stop = tf.compat.v1.placeholder(tf.float32, start_shape, 'stop')
```
[the remark this answers is not in the document]
Comment (rkazants) [reply tests/layer_tests/tensorflow_tests/test_tf_LinSpace.py:15]: Only scalars according to the spec: https://www.tensorflow.org/api_docs/python/tf/raw_ops/LinSpace

File: tests/layer_tests/tensorflow_tests/test_tf_LinSpace.py (hunk 1 above)
Comment (rkazants) [reply tests/layer_tests/tensorflow_tests/test_tf_LinSpace.py:24]: Only scalars supported

**Review by popovaan — APPROVED** [review]

**Review by mvafin — APPROVED** [review]

**Pull request merged**